File: Two pointers/leetcode_283.py

# Problem link ===>>  https://leetcode.com/problems/move-zeroes/description/
# submission link == =>> https://leetcode.com/problems/move-zeroes/submissions/1592649675/

from typing import List

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        l = 0
        n = len(nums)

        for r in range(n):
            if nums[r] != 0:
                nums[l], nums[r] = nums[r], nums[l]
                l += 1


if __name__ == "__main__":
    solution = Solution()

File: Two pointers/leetcode_283.cpp

// Problem link ===>> https://leetcode.com/problems/move-zeroes/description/

#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define endl '\n'

class Solution
{
public:
    void moveZeroes(vector<int> &nums)
    {
        int l = 0;
        int n = nums.size();
        
        for (int r = 0; r < n; r++)
        {
            swap(nums[l], nums[r]);
            if (nums[l])
                l++;
        }

        for (int i = 0; i < n; i++)
        {
            cout << nums[i] << " ";
        }
        cout << endl;
    }
};

int main()
{
    ios_base::sync_with_stdio(false);
    cout.tie(nullptr);
    cin.tie(nullptr);

    return 0;
}